### ISSUE TITLE
term_hash sigfault.

### ISSUE THREAD

**Issue by IanUtley** [SWI-Prolog/swipl-devel#1092]
```
?- term_hash(▄,X).
X = 14957940.

?- term_hash("▄",X).

SWI-Prolog [thread 1 (main) at Wed Jan 11 17:53:06 2023]: received fatal signal 11 (segv)
C-stack trace labeled "crash":
  [0] save_backtrace() at :? [0x7f3fab583f75]
  [1] print_c_backtrace() at :? [0x7f3fab5840c3]
  [2] sigCrashHandler() at :? [0x7f3fab5841da]
  [3] dispatch_signal() at :? [0x7f3fab4f3b20]
addr2line: DWARF error: section .debug_info is larger than its filesize! (0x93ef57 vs 0x530ea0)
  [4] killpg() at ??:? [0x7f3fab28e090]
  [5] MurmurHashAligned2() at :? [0x7f3fab522680]
  [6] primitiveHashValue___LD() at pl-termhash.c:? [0x7f3fab524cc6]
  [7] pl_term_hash2_va() at pl-termhash.c:? [0x7f3fab525fd5]
  [8] PL_next_solution___LD() at :? [0x7f3fab48d5a9]
  [9] query_loop() at :? [0x7f3fab4d84a5]
  [10] prologToplevel() at :? [0x7f3fab4d8d8f]
  [11] PL_toplevel() at ??:? [0x7f3fab55a3b2]
  [12] swipl(+0x10f5) [0x5565a4a0b0f5]
addr2line: DWARF error: section .debug_info is larger than its filesize! (0x93ef57 vs 0x530ea0)
  [13] __libc_start_main() at ??:? [0x7f3fab26f083]
  [14] swipl(+0x113e) [0x5565a4a0b13e]
Prolog stack:
  [10] system:term_hash/2 [PC=1 in supervisor]
  [9] $toplevel:toplevel_call/1 [PC=3 in clause 1]
  [8] $toplevel:stop_backtrace/1 [PC=4 in clause 1]
  [7] $tabling:$wfs_call/2 [PC=17 in clause 1]
  [5] $toplevel:$execute_goal2/3 [PC=29 in clause 1]
  [3] $toplevel:$query_loop/0 [PC=39 in clause 2]
  [2] $toplevel:$runtoplevel/0 [PC=19 in clause 1]
  [1] $toplevel:$toplevel/0 [PC=3 in clause 1]
  [0] system:$c_call_prolog/0 [PC=0 in top query clause]
Running on_halt hooks with status 139
Killing 3241 with default signal handlers
```

**Comment by kamahen** [comment]
Reproduced with `SWI-Prolog version 9.1.2` (built locally on Debian 11)

**Comment by JanWielemaker** [comment]
Thanks.  Fixed with f404856571496232020e7f763e8f546ecce74f84
